packages/x-date-pickers/src/internals/hooks/useField/useField.types.ts

```diff
@@ -102,6 +102,44 @@ export type UseFieldResponse<TForwardedProps extends UseFieldForwardedProps> = O
   };
 
 export interface FieldSection {
+  /**
+   * Value of the section, as rendered inside the input.
+   * For example, in the date `May 25, 1995`, the value of the month section is "May".
+   */
+  value: string;
+  /**
+   * Format token used to format generate the value of this section from the date object.
+   * For example, in the format `MMMM D, YYYY`, the format of the month section is "MMMM".
+   */
+  format: string;
+  /**
+   * Placeholder rendered when the value of this section is empty.
+   */
+  placeholder: string;
+  /**
+   * Name of the section.
+   */
+  name: MuiDateSectionName;
+  /**
+   * Type of content of the section.
+   * Will determine if we should apply a digit-based editing or a letter-based editing.
+   */
+  contentType: 'digit' | 'letter';
+  /**
+   * If `true`, the value of this section is supposed to have leading zeroes.
+   * For example, if the value `1` is rendered "01` instead of "1".
+   */
+  hasLeadingZeros: boolean;
+  /**
+   * If `true`, the section value has been modified since the last time the sections were generated from a valid date.
+   * When we can generate a valid date from the section, we don't directly pass it to `onChange`,
+   * Otherwise, we would lose all the information contained in the original date, things like:
+   * - time if the format does not contain it
+   * - timezone / UTC
+   *
+   * To avoid loosing those information, we transfer the values of the edited sections from the newly generated date to the original date.
```

Comment:
```suggestion
   * To avoid losing that information, we transfer the values of the edited sections from the newly generated date to the original date.
```